airflow/contrib/hooks/sqoop_hook.py

```diff
@@ -14,210 +14,282 @@
 #
 
 """
-This module contains a sqoop 1 hook
+This module contains a sqoop 1.x hook
 """
 
-from airflow.hooks.base_hook import BaseHook
-from airflow.exceptions import AirflowException
-
 import logging
 import subprocess
 
+from airflow.exceptions import AirflowException
+from airflow.hooks.base_hook import BaseHook
+
 log = logging.getLogger(__name__)
 
 
 class SqoopHook(BaseHook):
     """
     This Hook is a wrapper around the sqoop 1 binary. To be able to use te hook
     it is required that "sqoop" is in the PATH.
-    :param hive_home: (from json) The location of hive-site.xml
-    :type hive_home: str
-    :param job_tracker: (from json) <local|jobtracker:port> specify a job tracker
+    :param job_tracker: (from json) specify a job tracker local|jobtracker:port
     :type job_tracker: str
     :param namenode: (from json) specify a namenode
     :type namenode: str
-    :param lib_jars: (from json) specify comma separated jar files to
-        include in the classpath.
+    :param lib_jars: (from json) specify comma separated jar
+        files to include in the classpath.
     :type lib_jars: str
-    :param files: (from json) specify comma separated files to be copied
-        to the map reduce cluster
+    :param files: (from json) specify comma separated files to be copied to
+        the map reduce cluster
     :type files: (from json) str
     :param archives: (from json)  specify comma separated archives to be
         unarchived on the compute machines.
     :type archives: str
     """
-    def __init__(self, conn_id='sqoop_default'):
-        conn = self.get_connection(conn_id)
-        self.hive_home = conn.extra_dejson.get('hive_home', None)
-        self.job_tracker = conn.extra_dejson.get('job_tracker', None)
-        self.namenode = conn.extra_dejson.get('namenode', None)
-        self.lib_jars = conn.extra_dejson.get('libjars', None)
-        self.files = conn.extra_dejson.get('files', None)
-        self.archives = conn.extra_dejson.get('archives', None)
-        self.conn = conn
+
+    def __init__(self, conn_id='sqoop_default', verbose=False,
+                 num_mappers=None, properties=None):
+        # No mutable types in the default parameters
+        if properties is None:
+            properties = {}
+        self.conn = self.get_connection(conn_id)
+        connection_parameters = self.conn.extra_dejson
+        self.job_tracker = connection_parameters.get('job_tracker', None)
+        self.namenode = connection_parameters.get('namenode', None)
+        self.libjars = connection_parameters.get('libjars', None)
+        self.files = connection_parameters.get('files', None)
+        self.archives = connection_parameters.get('archives', None)
+        self.password_file = connection_parameters.get('password_file', None)
+        self.verbose = verbose
+        self.num_mappers = str(num_mappers)
```

Comment:
this line is wrong in the case you don't set `num_mappers`, it'll make `str(None)` and in the code then add to the command `--num_mappers None`

Reply:
Hey @Fokko thanks for the response! I also realized some of these things I mentioned you fixed in [this PR](https://github.com/apache/incubator-airflow/commit/4f459b64e098f4fd73af23030e7992e0465bfe21) I missed it originally, but still there are things that need to be fixed so I created [this PR](https://github.com/apache/incubator-airflow/pull/2703). I do need to add some tests but the gist is there. I've tested locally on my instance and made sure operators work how I expect.

But for instance this `num_mappers` issue is fixed in your PR for AIRFLOW-1217.